packages/data/src/factory.ts

```diff
@@ -33,17 +38,25 @@
  * registry as argument. The registry binding happens automatically when registering the selector
  * with a store.
  *
- * @param {Function} registrySelector Function receiving a registry `select`
- *                                    function and returning a state selector.
+ * @param registrySelector Function receiving a registry `select`
+ *                         function and returning a state selector.
  *
- * @return {Function} Registry selector that can be registered with a store.
+ * @return Registry selector that can be registered with a store.
  */
-export function createRegistrySelector( registrySelector ) {
+export function createRegistrySelector<
+	Selector extends ( ...args: any[] ) => any,
+>( registrySelector: ( select: typeof globalSelect ) => Selector ): Selector {
+	type WrappedSelector = {
```

Comment:
I think that type could make sense to be moved out of the inline context as it might be useful once we want to type the redux store package.

Reply:
Done.